compiler/rustc_driver/src/lib.rs

```diff
@@ -231,6 +231,8 @@ fn run_compiler(
         registry: diagnostics_registry(),
     };
 
+    init_rustc_env_logger_with_backtrace_option(&config.opts.unstable_opts.log_backtrace);
+
```

Comment:
I guess we'll stop getting logs before parsing command line flags, but that seems ok to me. The only alternative I see would be to use an env var instead of a command line flag, but I'm not sure if that matters much, so let's land it, we can always adjust it later if someone has a stronger opinion about it